### ISSUE TITLE
Login and better navigation drawer.

### ISSUE THREAD

**Issue by CytoDev** [jariz/Dumpert#14]
Starting this feature request to keep the [Rebrand](https://github.com/jariz/Dumpert/issues/11) thread clean.

**Comment by jariz** [comment]
Keep in mind that the UI for account management is already part of the library that is being used right now.
https://github.com/neokree/MaterialNavigationDrawer/wiki/Accounts

**Comment by CytoDev** [comment]
Dumpert does not use avatars. This would mean the clickable area in an account switcher is going to be confined to a spinner. Besides, an account switcher enables (and encourages) people to have/create multiple accounts. I would suggest handling accounts in a similar way the [reddit is fun](https://github.com/talklittle/reddit-is-fun) app by talklittle does.  
  
The first thing I'm going to do right now is create an activity to actually log in. The steps required to get to that activity is up for debate.

**Comment by CytoDev** [comment]
Logging in _seems_ to work if I believe my connection details at the moment. Dumpert just does not recognise my attempt as valid since it does not send back my commenter name and token. (Also checked and not banned from Dumpert)

logcat only outputs the message at [line 330](https://github.com/CytoDev/Dumpert/blob/dev/Dumpert/src/main/java/io/jari/dumpert/api/Login.java#L330).

**Comment by jariz** [comment]
Wat ik me kan herinneren is dat [dit gedeelte](https://github.com/jariz/GeenStijl/blob/master/src/io/jari/geenstijl/API/API.java#L267-L272) heel belangrijk was, want hij moest cookies automatisch accepteren en in de jar stoppen (ook tussen redirects etc.).
Heb je al geprobeerd die url waar je heen gestuurd wordt gewoon te volgen? Waarschijnlijk stuurt die je weer door naar app.steylloos.nl ?
Check in je browser de network tab, het is zeer waarschijnlijk dat er nog een redirect tussen zit.  
Het is echt een zooitje, het duurde ook behoorlijk lang voordat ik het werkend kreeg toen.
Meer dan dat kan ik er op dit moment ook niet aan toevoegen.

**Comment by CytoDev** [comment]
@jariz 
Redirects [worden gevolgd](https://github.com/CytoDev/Dumpert/blob/dev/Dumpert/src/main/java/io/jari/dumpert/api/Login.java#L304-L313) en cookies worden opgespaard... Alhoewel, er staan een aantal vreemde dingen op die regels die ik hoogstwaarschijnlijk niet met vol verstand heb getypt...

En ik had het zelfs nog kunnen zien als ik de logcat iets beter had bestudeerd:  
  
    01-05 16:31:36.517 2683-3409/io.jari.dumpert D/DLogin: setting email to: *************@*****.***
    01-05 16:31:36.517 2683-3409/io.jari.dumpert D/DLogin: setting password to: *********************
    01-05 16:31:36.517 2683-3409/io.jari.dumpert D/DLogin: created new cookiemonster
    01-05 16:31:36.517 2683-3409/io.jari.dumpert D/DLogin: connecting to: http://registratie.geenstijl.nl/registratie/gs_engine.php?action=login
                                                             using method: POST
                                                             using input: true
                                                             using output: true
                                                             using bindFormData: true
    01-05 16:31:36.828 2683-3409/io.jari.dumpert D/DLogin: Adding cookie: __cfduid=d7cd21d12ec581ec4500e65b5f76ea7341452007897; expires=Wed, 04-Jan-17 15:31:37 GMT; path=/; domain=.geenstijl.nl; HttpOnly
    01-05 16:31:36.830 2683-3409/io.jari.dumpert D/DLogin: Adding cookie: PHPSESSID=itg11j475omrlgvbir1lafgit4; path=/
    01-05 16:31:36.831 2683-3409/io.jari.dumpert D/DLogin: got responseCode: 302
    01-05 16:31:36.831 2683-3409/io.jari.dumpert D/DLogin: setting cookies to: __cfduid=d7cd21d12ec581ec4500e65b5f76ea7341452007897;PHPSESSID=itg11j475omrlgvbir1lafgit4
    01-05 16:31:36.831 2683-3409/io.jari.dumpert D/DLogin: connecting to: http://www.geenstijl.nl/registratie/index.php?view=login&__mode=handle_sign_in&t=666&_return=http%3A%2F%2Fapp.steylloos.nl%2Fmt-comments.fcgi%3F__mode%3Dhandle_sign_in%26static%3D1
                                                             using method: GET
                                                             using input: false
                                                             using output: false
                                                             using bindFormData: false
    01-05 16:31:36.831 2683-3409/io.jari.dumpert D/DLogin: found cookie. Using __cfduid=d7cd21d12ec581ec4500e65b5f76ea7341452007897;PHPSESSID=itg11j475omrlgvbir1lafgit4 as cookie
    01-05 16:31:36.979 2683-3409/io.jari.dumpert W/DLogin: No cookies found
    01-05 16:31:36.979 2683-3409/io.jari.dumpert D/DLogin: got responseCode: 302
    01-05 16:31:36.979 2683-3409/io.jari.dumpert D/DLogin: setting cookies to: __cfduid=d7cd21d12ec581ec4500e65b5f76ea7341452007897;PHPSESSID=itg11j475omrlgvbir1lafgit4
    01-05 16:31:36.979 2683-3409/io.jari.dumpert D/DLogin: connecting to: http://registratie.geenstijl.nl/registratie/index.php?view=login&__mode=handle_sign_in&t=666&_return=http%3A%2F%2Fapp.steylloos.nl%2Fmt-comments.fcgi%3F__mode%3Dhandle_sign_in%26static%3D1
                                                             using method: GET
                                                             using input: false
                                                             using output: false
                                                             using bindFormData: false
    01-05 16:31:36.979 2683-3409/io.jari.dumpert D/DLogin: found cookie. Using __cfduid=d7cd21d12ec581ec4500e65b5f76ea7341452007897;PHPSESSID=itg11j475omrlgvbir1lafgit4 as cookie
    01-05 16:31:37.463 2683-3409/io.jari.dumpert W/DLogin: No cookies found
    01-05 16:31:37.463 2683-3409/io.jari.dumpert D/DLogin: got responseCode: 200
    01-05 16:31:37.463 2683-3409/io.jari.dumpert W/DLogin: The server broke. Or user is a goldfish

**Comment by CytoDev** [comment]
Login functionality is working now. see [#11a8213](https://github.com/CytoDev/Dumpert/commit/11a82132c82ad327cfc77f2a3b6ad1e2ccbcf094).  
  
@jariz how do you want to implement the login UI and how do you want to use the functionality?  
Commenting on items has come one step closer. :tada:

**Comment by jariz** [comment]
@CytoDev Yea, they can do that on the GS site as well, I'd suggest taking out password reset and registration for now.
Do you want write access to this repo so we don't have to pass around PR's etc (you're going so fast, I can't keep up lol) because that would probably also make it easier to work from a separate branch together on the login feature.

**Comment by CytoDev** [comment]
@jariz Moved login to branch login. Removed button from master.

**Comment by CytoDev** [comment]
@jariz New navigation drawer is finished. Now it's just a matter of fixing the bugs that are left over by the old one, and implementing the login api from the login branch. I also removed the about section from the navigation drawer, and moved the settings to an overflow menu. I will put the about section in a sub-category in the preferences, along with a disclaimer and licenses.  
  
All in all, I gave myself too much time by saying I would be done by the end of the week. But then again, I won't be doing much with it tomorrow.
